hack/update-codegen.sh

```diff
@@ -449,46 +485,17 @@ openapi_definitions() {
 }
 export -f openapi_definitions
 
-if [[ $# -gt 0 && "$1" == "--parallel" ]]; then
-  shift 1
-  parallel --will-cite ::: \
-    authentication_groups \
-    core_groups \
-    extensions_groups \
-    resources_groups \
-    operator_groups \
-    seedmanagement_groups \
-    operations_groups \
-    settings_groups \
-    operatorconfig_groups \
-    controllermanager_groups \
-    admissioncontroller_groups \
-    scheduler_groups \
-    gardenlet_groups \
-    shoottolerationrestriction_groups \
-    shootdnsrewriting_groups \
-    provider_local_groups \
-    extensions_config_groups \
-    resourcemanager_groups
+parse_flags "$@"
+
+groups=($(get_groups $WHICH))
+if [[ "$MODE" == "sequential" ]]; then
+  for target in "${groups[@]}"; do
+    "$target"
+  done
+elif [[ "$MODE" == "parallel" ]]; then
+  parallel --will-cite ::: "${groups[@]}"
 else
-  authentication_groups
-  core_groups
-  extensions_groups
-  resources_groups
-  operator_groups
-  seedmanagement_groups
-  operations_groups
-  settings_groups
-  operatorconfig_groups
-  controllermanager_groups
-  admissioncontroller_groups
-  scheduler_groups
-  gardenlet_groups
-  resourcemanager_groups
-  shoottolerationrestriction_groups
-  shootdnsrewriting_groups
-  provider_local_groups
-  extensions_config_groups
+  printf "!! Invalid mode, Specify either 'parallel' or 'sequential'\n\n"
```

Comment:
Shouldn't with `exit 1` here? Otherwise the script can exit with 0 and give the wrong impression that it did the wanted job.